etna/records/templatetags/records_tags.py

```diff
@@ -36,7 +36,7 @@ def record_url(
 @register.simple_tag
 def is_page_current_item_in_hierarchy(page: Record, hierarchy_item: Record):
     """Checks whether given page matches item from a record's hierarchy"""
-    return page.reference_number == hierarchy_item.reference_number
+    return page.iaid == hierarchy_item.iaid
```

Comment:
Could be nice to have some test cases here, specially because of mentions like 
Re: 
> because item and sub-item seem to share the same reference number.

Reply:
Hi Allan, I've added a test for this but not sure how useful it is as it's only really comparing one value - it also comes from the loop inside the HTML which I think might be difficult to replicate here... :)

Reply:
Actually, I've updated this - ignore my comment :)